includes/plugins/class-woocommerce-subscriptions.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * WooCommerce Subscriptions integration class.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main class.
+ */
+class WooCommerce_Subscriptions {
+	/**
+	 * Initialize hooks and filters.
+	 */
+	public static function init() {
+		add_filter( 'woocommerce_subscription_settings', [ __CLASS__, 'add_on_hold_duration_setting' ], 11, 1 );
+		add_filter( 'wcs_default_retry_rules', [ __CLASS__, 'maybe_apply_on_hold_duration_rule' ], 99, 1 );
+	}
+
+	/**
+	 * Check if WooCommerce Subscriptions is active.
+	 *
+	 * @return bool
+	 */
+	public static function is_active() {
+		return class_exists( 'WC_Subscriptions' );
+	}
+
+	/**
+	 * Add on-hold duration setting.
+	 *
+	 * @param array $settings Subscription settings.
+	 *
+	 * @return array
+	 */
+	public static function add_on_hold_duration_setting( $settings ) {
+		if ( self::is_active() ) {
+			return array_merge(
+				$settings,
+				[
+					[
+						'name' => __( 'Newspack Subscriptions Settings', 'newspack-plugin' ),
+						'type' => 'title',
+						'desc' => __( 'Subscriptions settings added by Newspack.', 'newspack-plugin' ),
+						'id'   => 'newspack_subscriptions_options',
+					],
+					[
+						'name'              => __( 'On-hold Duration', 'newspack-plugin' ),
+						'id'                => 'newspack_subscriptions_on_hold_duration',
+						'css'               => 'max-width:80px;',
+						'value'             => self::get_on_hold_duration(),
+						'type'              => 'number',
+						'custom_attributes' => array(
+							'min'  => 0,
+							'step' => 1,
+						),
+						'desc'              => sprintf(
+							// Translators: %s is a line break.
+							__(
+								'Set the number of days a subscription remains in the On-Hold status after all automatic payment retries have failed.%sDuring this period, subscribers can update their payment details to restore the subscription at their current price. Once this time expires, the subscription will automatically transition to expired.',
+								'newspack-plugin'
+							),
+							'<br>'
+						),
+					],
+					[
+						'type' => 'sectionend',
+						'id'   => 'newspack_subscriptions_options',
+					],
+				]
+			);
+		}
+		return $settings;
+	}
+
+	/**
+	 * Get on-hold duration. Defaults to 0.
+	 *
+	 * @return int
+	 */
+	public static function get_on_hold_duration() {
+		return absint( get_option( 'newspack_subscriptions_on_hold_duration', 0 ) );
+	}
+
+	/**
+	 * Conditionally adds on-hold duration rule to retry rules.
+	 *
+	 * @param array $retry_rules Subscriptions retry rules.
+	 */
+	public static function maybe_apply_on_hold_duration_rule( $retry_rules ) {
+		if ( self::is_active() && count( $retry_rules ) > 0 ) {
```

Comment:
What are we checking with `count( $retry_rules )` ? In which case there will be no rules? If another plugin is tweaking it? Or if retries are disabled? If the latter, is there a more direct check for that option?

Reply:
> What are we checking with count( $retry_rules ) ? In which case there will be no rules? If another plugin is tweaking it? Or if retries are disabled? If the latter, is there a more direct check for that option?

I added the check mostly to account for the possibility of another plugin tweaking the rules. This particular hook would never be triggered if retries are disabled.

That said, I realize now that I had been assuming we enforce retries to be enabled for our sites. If this isn't the case, we would also need to force subscriptions to expire when this isn't enabled I think?

Reply:
> If this isn't the case, we would also need to force subscriptions to expire when this isn't enabled I think?

I don't think we enforce that. That's a good question. Let's open a ticket for that, as Katie and handle this in another PR if needed